apps/hash-ai-worker-ts/src/activities/infer-entities/propose-entities.ts

```diff
@@ -554,85 +477,182 @@ export const proposeEntities = async (params: {
-                          !inferenceState.proposedEntityCreationsByType[
-                            entityTypeId
-                          ]?.some(
-                            (existingEntity) =>
-                              existingEntity.entityId === entityId,
-                          ),
-                      )
-                      .map<ProposedEntity>(
+                return [
+                  entityTypeId,
+                  entities.filter(
+                    ({ entityId }) =>
+                      // Don't include invalid entities
+                      !invalidProposedEntities.some(
                         ({
-                          properties: simplifiedProperties,
-                          ...proposedEntity
-                        }) => {
-                          const { simplifiedPropertyTypeMappings } =
-                            entityTypes[entityTypeId] ?? {};
-
-                          if (!simplifiedPropertyTypeMappings) {
-                            throw new Error(
-                              `Could not find simplified property type mappings for entity type id ${entityTypeId}`,
-                            );
-                          }
-
-                          const properties = simplifiedProperties
-                            ? mapSimplifiedPropertiesToProperties({
-                                simplifiedProperties,
-                                simplifiedPropertyTypeMappings,
-                              })
-                            : {};
-
-                          return {
-                            ...proposedEntity,
-                            properties,
-                          };
-                        },
+                          invalidProposedEntity: { entityId: invalidEntityId },
+                        }) => invalidEntityId === entityId,
+                      ) &&
+                      // Ignore entities we've inferred in a previous iteration, otherwise we'll get duplicates
+                      !inferenceState.proposedEntityCreationsByType[
+                        entityTypeId
+                      ]?.some(
+                        (existingEntity) =>
+                          existingEntity.entityId === entityId,
                       ),
-                  ],
-                };
+                  ),
+                ];
               },
-              inferenceState.proposedEntityCreationsByType,
+            ),
+          );
+
+          const validProposedEntities = Object.values(
+            validProposedEntitiesByType,
+          ).flat();
+
+          const now = new Date().toISOString();
+
+          if (validProposedEntities.length > 0) {
+            logProgress(
+              typedEntries(validProposedEntitiesByType).flatMap(
+                ([entityTypeId, entities]) =>
+                  entities.map((entity) => ({
+                    proposedEntity: {
+                      ...entity,
```

Comment:
Can we un-simplify the properties here as well, so that we have the base URLs in the progress logs rather than the simple properties?

Reply:
Yes, forgot to do this when I merged in these changes. Have moved the "unsimplification" of properties to happen before they are logged in https://github.com/hashintel/hash/pull/4346/commits/69901f0d36896118a445d9e47dcede96ee26f939